cpp/oneapi/dal/test/engine/macro.hpp

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+// Disable clang-format as it dramatically
+// affects redability of macro definitions
+// clang-format off
+
+#define _TE_STRINGIFY(x) #x
```

Comment:
What are the rules for macro namings? When we add underscore prefix? Why we do not add `TE` prefix for all test system -related defines?

Reply:
Rule like "add prefix `TE` for all test system-related defines" will not work. There are macros from test test frameworks such as `ASSERT`, `TEST`, `GENERATE`, etc.

Here `_TE` is added to distinguish between "normal" macros such as `ASSERT`, just because I don't want them to be used explicitly in tests.